### ISSUE TITLE
Flycheck-flow does not work after upgrade

### ISSUE THREAD

**Issue by midsbie** [lbolla/emacs-flycheck-flow#32]
Flycheck-flow is not working at all when running the following Emacs and Flycheck versions:

GNU Emacs 25.1.1 (x86_64-pc-linux-gnu, GTK+ Version 3.22.11) of 2017-09-19, modified by Debian
Flycheck version: 32snapshot (package: 20171009.1145)

The error I see in the `*messages*` buffer is:

```
Error from syntax checker javascript-flow: Wrong type argument: flycheck-error-p, [cl-struct-flycheck-error #<buffer sessionstate.js> javascript-flow "/path/to/source.js" 17 22 "Property not found in" error "property `cite`"]
```

It worked fine before I upgraded packages from MELPA.  Anything I can do to help debugging this?

**Comment by lbolla** [comment]
Can you post a minimal example highlighting the issue?

**Comment by midsbie** [comment]
@lbolla: sample code that triggers an error:

```js
// @flow

export default function invalid(): string {
  return [];
}
```

When run from the command line, flow produces:
```
Error: index.js:4
 40:   return [];
              ^^ empty array literal. This type is incompatible with the expected return type of
 39: export default function invalid(): string {
                                        ^^^^^^ string


Found 1 error
```

From within emacs, the following error is raised:

Error from syntax checker javascript-flow: Wrong type argument: flycheck-error-p, [cl-struct-flycheck-error #<buffer test.js> javascript-flow "/path/to/test.js" 4 10 "This type is incompatible with the expected return type of" error "empty array literal"]

Thanks for looking into this!

**Comment by lbolla** [comment]
What version of `flow` are you using? It works for me with versions `0.31.0` and the latest `0.57.3`.

**Comment by midsbie** [comment]
I'm running flow `0.54.1` at the moment. So annoying. Will investigate further and see if it's maybe something related to my configuration.

**Comment by lbolla** [comment]
And what version of `flycheck`? I am running `flycheck-20171009.1145` from Melpa.

**Comment by midsbie** [comment]
Yes, same here; running the exact same version of flycheck and also from Melpa.

**Comment by lbolla** [comment]
I can see from
`Error from syntax checker javascript-flow: Wrong type argument: flycheck-error-p, [cl-struct-flycheck-error #<buffer sessionstate.js> javascript-flow "/path/to/source.js" 17 22 "Property not found in" error "property `cite`"]` that the parsing worked, but the filename is odd: `/path/to/source.js`.
Are you running the checker "on file save" or how else?

**Comment by midsbie** [comment]
No, I manually edited the file name. I can assure you that it is a correct and valid UNIX path name.  The checker is running dynamically as changes are made to the code. (I also have `prettier-mode`, which activates just before file save to prettify the buffer though I think it is unrelated)

I'm also `require`ing the `cl` Elisp library prior to setting up `flycheck` or `flycheck-flow` so I really don't understand how Emacs could be producing this error.  Might be related to the way I'm autoloading modes and such.

**Comment by lbolla** [comment]
Possibly. Try to deactivate everything but flycheck and flycheck-flow and see if it happens again.

**Comment by lbolla** [comment]
@miguel-guedes Any progress on this?

**Comment by midsbie** [comment]
Hi @lbolla! I haven't had the time to look into this but will close out the issue as it doesn't seem to be caused by Emacs Flow Flycheck.  Thanks for your help!
